### PR TITLE
OLM 4.11 release note: future PSA enforcement for SQLite catalogs

### PR DESCRIPTION
**Pull request by michaelryanpeter**
Version(s): 4.11

Issue: [OSDOCS-3774](https://issues.redhat.com//browse/OSDOCS-3774)

Link to docs preview (requires VPN):
[Future PSA enforcement for SQLite-based Operator catalogs](http://file.rdu.redhat.com/mipeter/olm-4.11-RN-opm-sqlite-catalogs-restricted-psas/release_notes/ocp-4-11-release-notes.html#ocp-4-11-olm-psa-opm-sqlite)

Additional information:
This extends the closed PR #48318. 

The work here and in [OSDOCS-3774](https://issues.redhat.com//browse/OSDOCS-3774) began as a known issue for 4.11. Upon further investigation, the pod security admission policy changes will affect catalog maintainers in a future OCP release, currently planned as 4.12.

cc: @dmesser @perdasilva @kevinrizza @kalexand-rh and @sjstout for visability.

### CODE REVIEW

**Comment by michaelryanpeter** [comment]
@jianzhangbjz or @Xia-Zhao-rh: Please review this PR. Thank you!

File: release_notes/ocp-4-11-release-notes.adoc [hunk 1]
```diff
@@ -629,6 +629,25 @@ The default Red Hat-provided Operator catalogs for {product-title} 4.11 releases
 
 For more information, see xref:../operators/understanding/olm-packaging-format.html#olm-file-based-catalogs_olm-packaging-format[File-based catalogs].
 
+[id="ocp-4-11-olm-psa-opm-sqlite"]
+==== Future pod security admission compatibility for SQLite-based Operator catalogs
```
Comment (jldohmann) [inline release_notes/ocp-4-11-release-notes.adoc:633]: If you want to shorten some of the sentences in this section, you could group the deprecation with its respective compatibility workaround (vs. deprecations grouped together and separate from grouped workarounds). E.g.

```
Operator catalogs that use the deprecated SQLite database format are not compatible with restricted namespace enforcement. It is recommended that catalog maintainers migrate to the new file-based catalog format to ensure compatibility.

Operator catalogs built with an `opm` CLI version earlier than 1.23.0 are not compatible with restricted namespace enforcement. It is recommended that catalog maintainers rebuild their SQLite-based catalogs using `opm` 1.23.0 or later to ensure compatibility.
```
Though it does introduce some redundancy in sentence structure and imo doesn't read as well as what you already have 🙂 though maybe there's a clever way to switch up some of the words that I can't see!

**Review by jldohmann — APPROVED** [review]
LGTM! I left a comment/suggestion below

**Comment by kalexand-rh** [comment]
@bandrade, can you take a look?

File: release_notes/ocp-4-11-release-notes.adoc [hunk 2]
```diff
@@ -629,6 +629,25 @@ The default Red Hat-provided Operator catalogs for {product-title} 4.11 releases
...
+
+The `restricted` pod security admission profile will be enabled for all namespaces in a future release of {product-title}.
+
+Operator catalogs built using both the deprecated SQLite database format and versions of the `opm` binary released before {product-title} 4.11 are not compatible with restricted namespace enforcement.
+
+To ensure compatibility with restricted namespace enforcement, maintainers of SQLite-based Operator catalogs can either migrate to the file-based catalog format or rebuild their catalogs using the `opm` binary released with {product-title} 4.11 or later.
```
Comment (jianzhangbjz) [inline release_notes/ocp-4-11-release-notes.adoc:639]: Does File-based Catalog Source compatible with restricted namespace? I guess no.
I deploy a file-based catalog source in a restricted namespace, and still get the PSA issue:
```yaml
/api/v1/namespaces/test/pods would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (container "registry-server" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "registry-server" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "registry-server" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "registry-server" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
/apis/batch/v1/namespaces/test/jobs would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (containers "util", "pull", "extract" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (containers "util", "pull", "extract" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or containers "util", "pull", "extract" must set securityContext.runAsNonRoot=true), seccompProfile (pod or containers "util", "pull", "extract" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```
And, I think the PSA issue nothing with the SQL-based or File-based, it's related to the Job pods: "registry-server", "util", "pull", "extract".
The File-based catalog source and restricted namespace.
```yaml
mac:~ jianzhang$ oc get catalogsource  test-operators -o yaml
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  creationTimestamp: "2022-08-01T02:34:22Z"
  generation: 1
  name: test-operators
  namespace: test
  resourceVersion: "88593"
  uid: fca4ab69-6cf4-4fbd-8318-182c72cdf297
spec:
  displayName: Jian Operators
  image: registry.redhat.io/redhat/redhat-operator-index:v4.11
  priority: -100
  publisher: Jian
  sourceType: grpc
  updateStrategy:
    registryPoll:
      interval: 10m0s
status:
  connectionState:
    address: test-operators.test.svc:50051
    lastConnect: "2022-08-01T02:38:47Z"
    lastObservedState: READY
  registryService:
    createdAt: "2022-08-01T02:34:22Z"
    port: "50051"
    protocol: grpc
    serviceName: test-operators
    serviceNamespace: test
mac:~ jianzhang$ oc get ns test -o yaml
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    openshift.io/description: ""
    openshift.io/display-name: ""
    openshift.io/requester: system:admin
    openshift.io/sa.scc.mcs: s0:c26,c15
    openshift.io/sa.scc.supplemental-groups: 1000680000/10000
    openshift.io/sa.scc.uid-range: 1000680000/10000
  creationTimestamp: "2022-08-01T02:32:18Z"
  labels:
    kubernetes.io/metadata.name: test
    olm.operatorgroup.uid/19192516-a967-48f5-a882-effe2a1e8fbb: ""
    pod-security.kubernetes.io/audit: privileged
    pod-security.kubernetes.io/audit-version: v1.24
    pod-security.kubernetes.io/warn: privileged
    pod-security.kubernetes.io/warn-version: v1.24
  name: test
  resourceVersion: "89724"
  uid: e9483d83-a011-4291-99ab-67c06345e3b2
spec:
  finalizers:
  - kubernetes
status:
  phase: Active
```

File: release_notes/ocp-4-11-release-notes.adoc [hunk 3]
```diff
@@ -629,6 +629,25 @@ The default Red Hat-provided Operator catalogs for {product-title} 4.11 releases
 
 For more information, see xref:../operators/understanding/olm-packaging-format.html#olm-file-based-catalogs_olm-packaging-format[File-based catalogs].
 
+[id="ocp-4-11-olm-psa-opm-sqlite"]
+==== Future pod security admission compatibility for SQLite-based Operator catalogs
+
+The `restricted` pod security admission profile will be enabled for all namespaces in a future release of {product-title}.
```
Comment (jianzhangbjz) [inline release_notes/ocp-4-11-release-notes.adoc:635]: It has been already enabled for 4.11 as default.

**Pull request merged**